dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/net/RequestExecutionContext.kt

```diff
@@ -0,0 +1,23 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.api.net
+
+/**
+ * Provides information about the current request execution context.
+ * @param attempts the number of attempts made to execute the request in case of a retry.
+ * @param previousResponseCode the response code of the previous request failure code in case of a retry.
+ */
+data class RequestExecutionContext(
+    /**
+     * The number of attempts made to execute the request in case of a retry.
+     */
+    val attempts: Int = 0,
+    /**
+     * The response code of the previous request failure code in case of a retry.
+     */
```

Comment:
why duplicate the description there if it is already in the Kdoc above? These ones won't be treated as documentation and are ordinary comments.